test/unit/resources/aws_s3_bucket_object_test.rb

```diff
@@ -0,0 +1,122 @@
+# encoding: utf-8
+require 'ostruct'
+require 'helper'
+require 'aws_s3_bucket_object'
+
+# MSBOSB = MockS3BucketObjectSingleBackend
+# Abbreviation not used outside this file
+
+#=============================================================================#
+#                            Constructor Tests
+#=============================================================================#
+class AwsS3BucketObjectConstructor < Minitest::Test
+  def setup
+    AwsS3BucketObject::BackendFactory.select(AwsMSBOSB::Basic)
+  end
+
+  def test_constructor_no_args_raises
+    assert_raises(ArgumentError) { AwsS3BucketObject.new }
+  end
+
+  def test_constructor_expected_well_formed_args
+    AwsS3BucketObject.new(name: 'Public Bucket', key: 'public_file.jpg')
+  end
+
```

Comment:
Please add a test showing that an exception will be thrown if only one of bucket_name or key is supplied.